pkg/util/client.go

```diff
@@ -117,3 +118,48 @@ func WatchFiles(candidates []string, callback func(fsnotify.Event)) error {
 
 	return nil
 }
+
+// LoadKubeConfigsV2 loads kubeconfigs. If the kubeconfigChangedCallBack is non-nil, it will watch all kubeconfigs it loaded
+// and call the callback once they change.
+func LoadKubeConfigsV2(opts flagutil.KubernetesOptions, kubeconfigChangedCallBack func()) (map[string]*rest.Config, error) {
+	configs, err := opts.LoadClusterConfigs(kubeconfigChangedCallBack)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load cluster configs: %w", err)
+	}
+	ret := map[string]*rest.Config{}
+	for k := range configs {
+		v := configs[k]
+		ret[k] = &v
+	}
+	return ret, nil
+}
+
+// WatchFilesV2 watches the passed files if they exist and calls callback for all events
+// except Chmod, as Openshift seems to generate frequent Chmod events
+func WatchFilesV2(candidates []string, callback func()) error {
```

Comment:
Doesn't the upstream code do this for us?

Reply:
Removed.